protocols/kad/src/protocol.rs

```diff
@@ -103,8 +103,11 @@ impl TryFrom<proto::Peer> for KadPeer {
 
         let mut addrs = Vec::with_capacity(peer.addrs.len());
         for addr in peer.addrs.into_iter() {
-            match Multiaddr::try_from(addr) {
-                Ok(a) => addrs.push(a),
+            match Multiaddr::try_from(addr).map(|addr| addr.with_p2p(node_id)) {
+                Ok(Ok(a)) => addrs.push(a),
+                Ok(Err(e)) => {
+                    log::debug!("Unable to parse multiaddr: {e} is not compatible with {node_id}");
```

Comment:
```suggestion
                Ok(Err(a)) => {
                    log::debug!("Unable to parse multiaddr: {a} is not compatible with {node_id}");
```

Given that it is an address. Not a strong opinion.

Reply:
I agree. It's changed now.